src/main/scala/com/snowplowanalytics/indicative/LambdaHandler.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2018-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.indicative
+
+// Scala
+import scala.collection.JavaConverters._
+
+// cats
+import cats.implicits._
+
+// AWS
+import com.amazonaws.services.lambda.runtime.events.KinesisEvent
+
+// This library
+import com.snowplowanalytics.indicative.Transformer.TransformationError
+import com.snowplowanalytics.snowplow.analytics.scalasdk.json.EventTransformer
+
+class LambdaHandler {
+
+  val apiKey: String =
+    sys.env.getOrElse("INDICATIVE_API_KEY",
+                      throw new RuntimeException("You must provide environment variable INDICATIVE_API_KEY"))
+
+  def recordHandler(event: KinesisEvent): Unit = {
+    val events = event.getRecords.asScala.map { record =>
+      val line      = new String(record.getKinesis.getData.array, "UTF-8")
+      val validated = EventTransformer.transformWithInventory(line)
+
+      validated
+        .leftMap(errors => TransformationError(errors.mkString("\n  * ")))
+        .flatMap(eventWithInventory => Transformer.transform(eventWithInventory.event, eventWithInventory.inventory))
+    }.toList
+
+    val (errors, jsons) = events.separate
+
+    errors.foreach(error => println("[Json transformation error]: " + error))
```

Comment:
can we have those printlns in `IO` and have only one big `IO` that we run at the end?